services/web-app/components/utils/useOutsideClick.js

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright IBM Corp. 2021, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { useEffect } from 'react'
+
+const useOutsideClick = (ref, callback) => {
+  const handleClick = (e) => {
+    if (ref.current && !ref.current.contains(e.target)) {
+      callback()
+    }
+  }
+
+  useEffect(() => {
```

Comment:
Should this `useEffect` have an empty dependency array `[]` to prevent it from running too much? If not, let's add a comment to clarify this as it's expected that `useEffect`s have stated dependencies.